subtests/docker_cli/cp/cp.py

```diff
@@ -207,3 +208,72 @@ def verify_files_number(self, copied_number, expected_number):
         self.loginfo("Success, copied %d files from container, "
                      "expected number from configuration %d"
                      % (copied_number, expected_number))
+
+
+class volume_mount(CpBase):
+    """
+    Regression between docker-1.10.3 and 1.12.
+
+    See: https://github.com/docker/docker/issues/27773
+    """
+
+    def initialize(self):
+        super(volume_mount, self).initialize()
+
+        vol = 'v_' + utils.generate_random_string(8)
+        subargs = ['create', '--name', vol]
+        mustpass(DockerCmd(self, 'volume', subargs).execute())
+        self.sub_stuff['volume_name'] = vol
+
+        # First container: bind-mounts new volume, then unpacks two
+        # tarballs into it. I don't know why it has to be two tarballs
+        # and not one, but the tar thing (vs plain copy) has to do
+        # with pivot_root. Whatever that is.
+        c1 = 'c1_' + utils.generate_random_string(8)
+        # Path is not configurable because it's hardcoded in the tarballs
+        vol_binding = vol + ':/.imagebuilder-transient-mount'
+        subargs = ['--name', c1, '-v', vol_binding, 'busybox']
+        mustpass(DockerCmd(self, 'create', subargs).execute())
+        self.sub_stuff['container1'] = c1
+        for ab in ['a', 'b']:
+            tar_file = 'cp_volume_mount_data_%s.tar' % ab
+            tar_path = os.path.join(self.parent_subtest.bindir, tar_file)
+            # Failure mode only happens if cp is via redirected stdin.
```

Comment:
redirected ``stdin`` == ``AsyncDockerCmd.execute(_int_)`` where_int_ is the ``r`` side from ``os.pipe()``.  Docker will block until you start filling the ``w`` side, and it will complete when you close it....

...Yeah, ``utils.run()`` is probably a whole lot easier.  

You should use ``self.config[docker_path]`` though, b/c at some point we may be testing multiple clients :D